pkg/deployment/deployment.go

```diff
@@ -163,73 +174,55 @@ func (p *PartRole) UnmarshalJSON(data []byte) (err error) {
-	MountPoint  string     `json:"mountPoint,omitempty"`
-	MountOpts   []string   `json:"mountOpts,omitempty"`
-	RWVolumes   RWVolumes  `json:"rwVolumes,omitempty"`
-	UUID        string     `json:"uuid,omitempty"`
+	Label       string     `yaml:"label,omitempty"`
+	FileSystem  FileSystem `yaml:"fileSystem,omitempty"`
+	Size        MiB        `yaml:"size,omitempty"`
+	Role        PartRole   `yaml:"role"`
+	StartSector uint       `yaml:"startSector,omitempty"`
+	MountPoint  string     `yaml:"mountPoint,omitempty"`
+	MountOpts   []string   `yaml:"mountOpts,omitempty"`
+	RWVolumes   RWVolumes  `yaml:"rwVolumes,omitempty"`
+	UUID        string     `yaml:"uuid,omitempty"`
 }
 
 type Partitions []*Partition
 
 type Disk struct {
-	Device      string     `json:"device,omitempty"`
-	Partitions  Partitions `json:"partitions"`
-	StartSector uint       `json:"startSector,omitempty"`
+	// omit the device name as this is a runtime information which might
+	// not be consistent across reboots, there is no need to store it.
+	Device      string     `yaml:"-"`
+	Partitions  Partitions `yaml:"partitions"`
+	StartSector uint       `yaml:"startSector,omitempty"`
 }
 
 type BootConfig struct {
-	Bootloader    string `json:"name"`
-	KernelCmdline string `json:"kernelCmdline"`
-}
-
-// MarshalJSON on disks omits the device name as this is a runtime information
-// which might not be consistent across reboots, there is no need to store it.
-func (d Disk) MarshalJSON() ([]byte, error) {
-	type diskAlias Disk
-	disk := diskAlias(d)
-	disk.Device = ""
-	return json.Marshal(disk)
+	Bootloader    string `yaml:"name"`
+	KernelCmdline string `yaml:"kernelCmdline"`
 }
 
 type FirmwareConfig struct {
-	BootEntries []*firmware.EfiBootEntry `json:"entries"`
+	BootEntries []*firmware.EfiBootEntry `yaml:"entries"`
 }
 
 type Deployment struct {
-	SourceOS   *ImageSource    `json:"sourceOS"`
-	Disks      []*Disk         `json:"disks"`
-	Firmware   *FirmwareConfig `json:"firmware"`
-	BootConfig *BootConfig     `json:"bootloader"`
+	SourceOS   *ImageSource    `yaml:"sourceOS"`
+	Disks      []*Disk         `yaml:"disks"`
+	Firmware   *FirmwareConfig `yaml:"firmware"`
+	BootConfig *BootConfig     `yaml:"bootloader"`
 	// Consider adding a systemd-sysext list here
 	// All of them would extracted in the RO context, so only
 	// additions to the RWVolumes would succeed.
-	OverlayTree *ImageSource `json:"overlayTree,omitempty"`
-	CfgScript   string       `json:"configScript,omitempty"`
-}
-
-// MarshalJSON on deploy omits the overlay tree and the configuration script
-// as these are install|ugrade time information which is no longer guaranteed
-// to be (re)applicable on a deployed system as those are not kept aside.
-func (d Deployment) MarshalJSON() ([]byte, error) {
-	type deployAlias Deployment
-	deploy := deployAlias(d)
-	deploy.CfgScript = ""
-	deploy.OverlayTree = nil
-	return json.Marshal(deploy)
+	OverlayTree *ImageSource `yaml:"-"`
+	CfgScript   string       `yaml:"-"`
```

Comment:
Isn't this a behavior change? Now this can only be set from the command line flags, not as part of a config yaml.

Reply:
Fixed in commit dc7da30a609950475c635da1048ad5fff397c142